### PR TITLE
Test actions interests

### PR DESCRIPTION
**Pull request by cappers86**
# Description

Add tests to interests 

## Checklist

Remove any items which are not applicable.

- [x] I have performed a self-review of my own code

## Linkt to Trello card

> https://trello.com/c/1hJ6oh6K/45-add-tests-for-actions-in-interestsjs

### CODE REVIEW

**Comment by cappers86** [comment]
[Add tests for actions in interests.js](https://trello.com/c/1hJ6oh6K/45-add-tests-for-actions-in-interestsjs)

File: commit_template.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-commit message format
+():commit message format
```
Comment (evansibok) [inline commit_template.md]: I'll like to know the purpose of this change please. Doesn't seem like it has anything to do with the tests.

File: src/state/actions/__test__/avgSalaries.test.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+import { getAvgSalaries } from '../avgSalaries';
+
+const middlewares = [thunk];
+const mockStore = configureStore(middlewares);
+const mock = new MockAdapter(axios);
+
+const avgSalaries = [
+  {
+    interest_id: 2,
```
Comment (evansibok) [inline src/state/actions/__test__/avgSalaries.test.js]: The `interest_id` only exists on the users DB but isn't currently being used in the app, so this test doesn't need that property anymore. Our current work on the salaries takes out this property. When the salaries branch has been merged, you can `git pull origin develop` and` knex migrate:down` your DB till you get to the last table, then redo `knex migrate:latest` and `knex seed:run` on both the backend development and testing databases, then start up your tests again.

File: src/state/actions/__test__/interests.test.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+import { getUsersInterests, getInterests } from '../interests';
+
+const middlewares = [thunk];
+const mockStore = configureStore(middlewares);
+const mock = new MockAdapter(axios);
+
+const interests = {
+  id: 1,
+  interest: 'AI Engineer',
+};
```
Comment (evansibok) [inline src/state/actions/__test__/interests.test.js]: The `interests` endpoint is supposed to return an array of objects and not an `object`.

File: src/state/actions/__test__/interests.test.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+      type: types.GET_INTERESTS_SUCCESS,
+      payload: interests,
+    };
+    const store = mockStore({});
+    const actions = store.getActions();
+    await store.dispatch(getInterests(1));
+    expect(actions[1]).toEqual(expectedActions);
+  });
+  it('should execute fetch error data', async () => {
+    const code = 401;
```
Comment (evansibok) [inline src/state/actions/__test__/interests.test.js]: When a get request fails, it's meant to return a `404`, I'm not sure why it's returning a `401`. A `401` should be representing `unauthorized`. So it seems the `getInterests` function is only accounting for when the user doesn't have access and not accounting for when the resource fetched isn't gotten or there's an error in the request. I think this is something we can look at in the future when modifying the `getInterests` action file. That's if we end up using it. LOL

File: src/state/actions/__test__/interests.test.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    mock
+      .onGet(`${process.env.REACT_APP_BACKEND_URL}/interests/user/1`)
+      .reply(code2, { message: 'Request failed with status code 404' });
+    const expectedAction = {
+      type: types.GET_USER_INTERESTS_FAILURE,
+      payload: `Request failed with status code ${code2}`,
+    };
+    const store = mockStore({});
+    const actions = store.getActions();
+    await store.dispatch(getUsersInterests());
```
Comment (evansibok) [inline src/state/actions/__test__/interests.test.js]: The `getUsersInterests` action takes an `id` parameter but it's not being account for here. I'm not sure why the test still passes though. I was of the opinion that if the id parameter isn't passed in the test mock it should fail. Hmmm

**Pull request merged**